src/modules/cpu/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,842 @@
+            int index = batchCount * 4;
+            tensorSumArr[index] = (Rpp64u)sumR;
+            tensorSumArr[index + 1] = (Rpp64u)sumG;
+            tensorSumArr[index + 2] = (Rpp64u)sumB;
+            tensorSumArr[index + 3] = sum;
+        }
+
+    }
+
+    return RPP_SUCCESS;
+}
+
+RppStatus tensor_sum_f32_f32_host(Rpp32f *srcPtr,
+                                  RpptDescPtr srcDescPtr,
+                                  Rpp32f *tensorSumArr,
+                                  RpptROIPtr roiTensorPtrSrc,
+                                  RpptRoiType roiType,
+                                  RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(srcDescPtr->n)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f *srcPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp32f *srcPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+
+        Rpp32u alignedLength = (bufferLength / 24) * 24;
+        Rpp32u vectorIncrement = 24;
+        Rpp32u vectorIncrementPerChannel = 8;
+
+        // Tensor Sum without fused output-layout toggle (NCHW)
+        if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+        {
+            alignedLength = bufferLength & ~7;
+            vectorIncrement = 8;
+            Rpp64f sum = 0.0;
+            Rpp64f sumAvx[4] = {0.0};
+
+            Rpp32f *srcPtrRow;
+            srcPtrRow = srcPtrChannel;
+#if __AVX2__
+            __m256d psum = _mm256_setzero_pd();
+#endif
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp32f *srcPtrTemp;
+                srcPtrTemp = srcPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrement)
+                {
+#if __AVX2__
+                    __m256d p1[2];
+                    rpp_simd_load(rpp_load8_f32_to_f64_avx, srcPtrTemp, p1);
+                    compute_sum_8_host(p1, &psum);
+#endif
+                    srcPtrTemp += vectorIncrement;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    sum += (Rpp64f)(*srcPtrTemp);
+                    srcPtrTemp++;
+                }
+                srcPtrRow += srcDescPtr->strides.hStride;
+            }
+#if __AVX2__
+            rpp_simd_store(rpp_store4_f64_to_f64_avx, sumAvx, &psum);
+#endif
+            for(int i = 0; i < 2; i++)
```

Comment:
please unroll the loop here

Reply:
Unrolled the loop here and in all other places.